### ISSUE TITLE
Cannot use with TypeScript target ES5

### ISSUE THREAD

**Issue by cloutiertyler** [jacobwgillespie/styled-icons#162]
I'm having difficulties importing the styled icons because it appears that the generated files are using ES6 exporting. It looks like your `module` setting in your tsconfig is `commonjs` so I'm not sure why that would be.

In particular the error is:
```
(function (exports, require, module, __filename, __dirname) { export { AddressBook } from './AddressBook';
                                                              ^^^^^^

SyntaxError: Unexpected token export
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:607:28)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
```

**Comment by jacobwgillespie** [comment]
Hey, how are you bundling your application?  The normal files are built as ES modules (specifically `target: es5` and `module: es2015`) and it's expected that your JS bundler is able to do the transformation from ES modules to CommonJS, etc.  That should work with Webpack, Rollup, etc. depending on how they're configured.

The benefit is tree shaking, as if you are only going to use a few icons, you probably don't want to bundle the entire icon set.

If it's not possible to configure your bundler to understand ES modules, there's also a CommonJS version of every file available with the suffix `.cjs`.  For example:

```typescript
// The default and preferred import
import {AddressBook} from 'styled-icons/fa-regular'

// The CommonJS import (although this bundles the _entire_ FA icon set
import {AddressBook} from 'styled-icons/fa-regular/index.cjs'

// To only get the icons you need
import {AddressBook} from 'styled-icons/fa-regular/AddressBook.cjs'
```

Can you give me some more details on your bundler configuration?  Webpack version?  A gist with the config?

**Comment by peterlazzarino** [comment]
I am seeing the same issue with an installation into a gatsby react app

**Comment by jacobwgillespie** [comment]
@peterlazzarino do you have a repo that reproduces the issue?  I can take a look.  I suspect that Gatsby might require the `.cjs` extension for Gatsby 1.x and should not require it for 2.x (the most recent release looks like `2.0.0-beta.19`).  The 2.x branch upgraded to Webpack 4, but the current Gatsby 1.x releases use Webpack 1.

However, having said that, the Styled Icons website in this repo is itself built with Gatsby 1.9 and doesn't seem to require `.cjs`, so I'd be happy to take a look at your setup.

**Comment by peterlazzarino** [comment]
@jacobwgillespie my main dev repo is private but I created a basic public repo that reproduces this issue

https://github.com/peterlazzarino/gatsby-test

Just run it and you will see the issue preventing app render.

The repo is based on the gatsby-netlify-starter that my app is also built on top of.

**Comment by RXminuS** [comment]
I have the same problem but my target is set to ES6. I tried both the cjs and non-cjs methods.

```

styles.ts:13 Connecting to fusebox HMR at ws://localhost:4445
.../app-ts/node_modules/styled-icons/fa-regular/AddressBook.js:5 Uncaught SyntaxError: Unexpected token import
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:606:28)
    at Object.Module._extensions..js (module.js:653:10)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:504:12)
    at Function.Module._load (module.js:496:3)
    at Module.require (module.js:586:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (.../node_modules/styled-icons/fa-regular/index.cjs.js:3:21)
```

**Comment by RXminuS** [comment]
<img width="465" alt="screen shot 2018-07-11 at 16 37 05" src="https://user-images.githubusercontent.com/3949285/42579474-b10bbd84-8528-11e8-805f-ac5a1491619b.png">

**Comment by peterlazzarino** [comment]
To follow up, and as a reply to the comment above, the .cjs method works in my app.

**Comment by peterlazzarino** [comment]
@RXminuS can you post the pre-compile version of your component?

**Comment by RXminuS** [comment]
Could this have something to do with it?
https://stackoverflow.com/questions/42685501/confusing-typescript-errors-due-to-hyphen-in-package-name

**Comment by jacobwgillespie** [comment]
Okay, I believe I understand what's going on:

* Imports like `styled-icons/material` or `styled-icons/material/Person` are ES5 but use ES modules (`import` / `export`) in order to support tree shaking, which is a good idea given that the import can be very large if all of the icons end up being bundled unintentionally
* Most bundlers like Webpack, Rollup, Parcel, etc should natively understand the ES modules and utilize them for tree shaking
* Gatsby 1.x uses Webpack 1, which is outdated and does not understand ES modules
* Gatsby 2.x uses Webpack 4 which should just work out of the box, however it's still in beta
* As a temporary workaround, adding the `.cjs` suffix should work: `styled-icons/material/Person.cjs`
* This repo uses Gatsby 1.x, however since the files are "local" and aren't inside `node_modules`, they are converted to CommonJS by TypeScript before building, which is why it works here and won't work when `styled-icons` is a dependency

@peterlazzarino thanks for the repo, I was able to reproduce with it.

@RXminuS I don't think it's due to the hyphen - do you have a repo I could take a look at?  From your error message, it looks like you're using FuseBox as the bundler, which I have not yet tested.  It should be similar to Webpack, but I'd be happy to take a look if you have a sample repo.

**Comment by craigmd** [comment]
To elaborate on what @RXminuS reported. The `index.cjs` file is trying to `require` the non cjs version and again webpack is kicking back an error saying it doesn't know what `import` is. So,
```
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ThreeDRotation_1 = require("./ThreeDRotation.cjs"); // works
exports.ThreeDRotation = ThreeDRotation_1.ThreeDRotation;
var Accessibility_1 = require("./Accessibility"); // error
```

**Comment by jacobwgillespie** [comment]
Ah nice catch, yup that's a bug.  It should be fixed in #192 and released as v2.3.1.

**Comment by jacobwgillespie** [comment]
This should be resolved:

* If you're using Webpack 3/4 / Create React App / Rollup the default entry points should work and should support tree-shaking
* Otherwise, the `.cjs` suffixed entry points should work, at the risk of making tree shaking more difficult

Feel free to open a new issue if anything else comes up!  🎉
